tidy3d/material_library/parametric_materials.py

```diff
@@ -0,0 +1,266 @@
+"""Parametric material models."""
+from abc import ABC, abstractmethod
+from typing import List, Tuple
+import pydantic as pd
+import numpy as np
+from scipy import integrate
+
+from ..components.medium import PoleResidue, Medium2D, Drude
+from ..components.base import Tidy3dBaseModel
+from ..constants import EPSILON_0, Q_e, HBAR, K_B
+
+GAMMA_DEF = 0.11e-3
+MU_C_DEF = 0.3
+TEMP_DEF = 300
+
+MIN_FREQ_THZ = 1
+MAX_FREQ_THZ = 1000
+NUM_FREQS = 1000
+
+
+class ParametricVariantItem2D(ABC, Tidy3dBaseModel):
+    """A variant of a 2D material depending on parameters, that must be initialized in order to
+    generate the material model."""
+
+    @property
+    @abstractmethod
+    def medium(self) -> Medium2D:
+        """Calculate the material model at the current parameter values."""
+
+
+class Graphene(ParametricVariantItem2D):
+    """Surface conductivity model for graphene, including intraband and interband terms,
+    as described in
+
+    George W. Hanson, "Dyadic Green's Functions for an Anisotropic,
+    Non-Local Model of Biased Graphene," IEEE Trans. Antennas Propag.
+    56, 3, 747-757 (2008).
+
+    Example
+    -------
+    >>> medium = Graphene(mu_c = 0.2).medium
+
+    """
+
+    gamma: float = pd.Field(
+        GAMMA_DEF,
+        title="Scattering rate in eV",
+        description="Scattering rate in eV.",
+    )
+    mu_c: float = pd.Field(
+        MU_C_DEF,
+        title="Chemical potential in eV",
+        description="Chemical potential in eV.",
+    )
+    temp: float = pd.Field(
+        TEMP_DEF,
+        title="Temperature in K",
+        description="Temperature in K.",
+    )
+    include_interband: bool = pd.Field(
+        True,
+        title="Include interband terms",
+        description="Include interband terms, relevant at high frequency (IR). "
+        "Otherwise, the intraband terms only give a simpler Drude-type model relevant "
+        "only at low frequency (THz).",
+    )
+
+    @property
+    def medium(self) -> Medium2D:
+        """Surface conductivity model for graphene."""
+        intraband = self.drude_intraband
+        if self.include_interband:
+            interband = self.pole_residue_interband
+            intraband_poles = intraband.pole_residue.poles
+            interband_poles = interband.pole_residue.poles
```

Comment:
duplicate line of code?

Reply:
no, one is intraband and one is interband, then we add them

Reply:
oh I see now. looked the same upon first glance